usr/sbin/truenas-install

```diff
@@ -282,8 +289,18 @@ install_loader()
 	    chroot ${_mnt} grub-install --target=i386-pc /dev/${_disk}
 
 	    echo "Stamping EFI loader on: ${_disk}"
-	    chroot ${_mnt} mkdosfs -F 32 -s 1 -n EFI /dev/${_disk}2
+        # Check if nvme disk and use different naming convention.
+        if [[ "${_disk}" == *"nvme"* ]]; then
+            _partition="${_disk}p3"
+        else
+            _partition="${_disk}3"
+        fi
+	    chroot ${_mnt} mkdosfs -F 32 -s 1 -n EFI /dev/${_partition}
+	    chroot ${_mnt} mount -t vfat /dev/${_disk}p2 /boot/efi
+        else
+        chroot ${_mnt} mkdosfs -F 32 -s 1 -n EFI /dev/${_partition}
 	    chroot ${_mnt} mount -t vfat /dev/${_disk}2 /boot/efi
```

Comment:
We still don't have consistent spaces, please see the line below this.

Reply:
Not entirely sure what spaces you're referring to but that 'else' shouldn't be there, nor that another line of code below it. I'm just going to start from the source file again, and this time make a function.

Reply:
The `else` definitely shouldn't be there. The entire block should be fixed after `fi`.